share/p2p/discovery/discovery.go

```diff
@@ -195,26 +199,34 @@ func (d *Discovery) Advertise(ctx context.Context) {
 }
 
 // discoveryLoop ensures we always have '~peerLimit' connected peers.
-// It starts peer discovery per request and restarts the process until the soft limit reached.
+// It initiates peer discovery upon request and restarts the process until the soft limit is reached.
 func (d *Discovery) discoveryLoop(ctx context.Context) {
 	t := time.NewTicker(discoveryRetryTimeout)
 	defer t.Stop()
+
+	warnTicker := time.NewTicker(logInterval)
+	defer warnTicker.Stop()
+
 	for {
-		// drain all previous ticks from channel
+		// drain all previous ticks from the channel
 		drainChannel(t.C)
 		select {
 		case <-t.C:
 			found := d.discover(ctx)
+
+			// Added: Print out the number of peers discovered.
+			log.Infof("Number of peers discovered: %d Required: %d", d.set.Size(), d.set.Limit())
```

Comment:
Were you wanting to make add this info to new Warn log line? 
It would be better to add discovered and required size into new Warn line and remove this one. Logging here could result in spammy INFO logs, since this code could be retried quite frequently.